packages/driver/src/cy/keyboard.ts

```diff
@@ -260,7 +260,11 @@ const shouldUpdateValue = (el: HTMLElement, key: KeyDetails, options) => {
 
       if (!(numberRe.test(potentialValue))) {
         debug('skipping inserting value since number input would be invalid', key.text, potentialValue)
-        options.prevVal = needsValue + key.text
+        if (key.text.match(/[-+eE\d\.]/)) {
```

Comment:
this `regexp` should be hoisted to the top and given a variable name to help describe what it is

Reply:
also a description would be helpful